atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -737,31 +738,32 @@ private Comparator<Cell> preserveInputRowOrder(List<Map.Entry<Cell, Value>> inpu
     private Iterator<Map.Entry<byte[], RowColumnRangeIterator>> partitionByRow(
             Iterator<Map.Entry<Cell, Value>> rawResults, Iterator<byte[]> expectedRows) {
         PeekingIterator<Map.Entry<Cell, Value>> peekableRawResults = Iterators.peekingIterator(rawResults);
-        return new AbstractIterator<Map.Entry<byte[], RowColumnRangeIterator>>() {
+        return new AbstractIterator<>() {
             byte[] prevRowName;
 
             @Override
             protected Map.Entry<byte[], RowColumnRangeIterator> computeNext() {
                 finishConsumingPreviousRow(peekableRawResults);
                 if (!expectedRows.hasNext()) {
-                    Preconditions.checkState(
-                            !peekableRawResults.hasNext(),
-                            "Iterators are not consistent: there is some data returned by getRowsColumnRange()"
-                                    + " even when we expect no more rows. This is likely an AtlasDB product bug.",
-                            UnsafeArg.of("nextRawResult", peekableRawResults.peek()));
+                    if (peekableRawResults.hasNext()) {
+                        throw new SafeIllegalStateException(
+                                "Iterators are not consistent: there is some data returned by getRowsColumnRange()"
+                                        + " even when we expect no more rows. This is likely an AtlasDB product bug.",
+                                UnsafeArg.of("nextRawResult", peekableRawResults.peek()));
```

Comment:
oof, yeah I added the unsafe-arg as an afterthought (and being a logically impossible case right now didn't test it!)